### PR TITLE
Add 1 min safety window for commit listing

### PR DESCRIPTION
**Pull request by nitishkumar71**
Add 1 min safety window for commit listing
## Description

building a new function includeCommits() based includePR()
which adds an 1 minute safety window for commit listing.
This will help Derek to avoid including commits from prior
releases.

## Motivation and Context

- [x] Add 1 min safety window for commit listing #143 ([required](https://github.com/alexellis/derek/issues/143))

## How Has This Been Tested?

Run `./build.sh` to run test cases and build Docker image 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Thank you for this PR, I would like to go ahead and merge it, but wanted to see if you could test it end to end? If it's not tested, then I'd be releasing a change into the live system which may not work as expected and could cause confusion to some of our users.

Setting up the GitHub App for Derek can take 10-15 minutes the first time you do it, but from there you can use Ngrok or https://inlets.dev/ to get a public IP and receive a Webhook.

What do you think?

Alex

File: handler/release_handler.go [hunk 1]
```diff
@@ -204,3 +206,13 @@ func updateRelease(client *github.Client, release *github.RepositoryRelease, own
 	_, _, err := client.Repositories.EditRelease(context.Background(), owner, repo, *release.ID, release)
 	return err
 }
+
+func includeCommits(cm github.RepositoryCommit, previous, current time.Time) bool {
```
Comment (alexellis) [inline handler/release_handler.go]: I think this should be called `includeCommit` given that you are working on a single item at a time?

Comment (nitishkumar71) [reply handler/release_handler.go]: Agree, will make that change.

**Comment by nitishkumar71** [comment]
> Thank you for this PR, I would like to go ahead and merge it, but wanted to see if you could test it end to end? If it's not tested, then I'd be releasing a change into the live system which may not work as expected and could cause confusion to some of our users.
> 
> Setting up the GitHub App for Derek can take 10-15 minutes the first time you do it, but from there you can use Ngrok or https://inlets.dev/ to get a public IP and receive a Webhook.
> 
> What do you think?
> 
> Alex
Sure @alexellis, sounds good to me.

Just need clarification on, How should we document the result of the end to end test?

File: handler/release_handler.go (hunk 1 above)
Comment (nitishkumar71) [reply handler/release_handler.go]: Done

**Comment by burtonr** [comment]
@nitishkumar71 for the end to end tests, you could either link to a sample repo where you have your updated version of Derek running, or you could add some screenshots of the updated Derek code working like what I've done in the past: https://github.com/alexellis/derek/pull/104#issuecomment-440930941

Basically, we just need to see that it works properly in a real scenario. A link, or a screenshot would be enough to know that it's doing what it's intended to do.

If you need help getting Derek setup and running, you can ask in the `#derek` channel in the OpenFaaS Slack

**Comment by alexellis** [comment]
Thank you @burtonr 👍 

I have a feeling that we will need to cover this, so will take it on and report back if there are any bugs / issues.

**Comment by nitishkumar71** [comment]
> @nitishkumar71 for the end to end tests, you could either link to a sample repo where you have your updated version of Derek running, or you could add some screenshots of the updated Derek code working like what I've done in the past: [#104 (comment)](https://github.com/alexellis/derek/pull/104#issuecomment-440930941)
> 
> Basically, we just need to see that it works properly in a real scenario. A link, or a screenshot would be enough to know that it's doing what it's intended to do.
> 
> If you need help getting Derek setup and running, you can ask in the `#derek` channel in the OpenFaaS Slack


Got it @burtonr, will do it in some time.

**Pull request merged**